com.civicrm.hrjobroles/tests/phpunit/CRM/Hrjobroles/Import/Parser/HrJobRolesTest.php

```diff
@@ -0,0 +1,125 @@
+
+require_once 'HrJobRolesTestBase.php';
+
+class CRM_Hrjobroles_Import_Parser_HrJobRolesTest extends HrJobRolesTestBase {
+
+
+  function setUp() {
+    parent::setUp();
+    $session = CRM_Core_Session::singleton();
+    $session->set('dateTypes', 1);
+    $this->createSampleOptionGroupsAndValues();
+  }
+
+  function tearDown() {
+    parent::tearDown();
+  }
+
+  function testBasicImport() {
+    // create contact
+    $contactParams = ['first_name'=>'walter', 'last_name'=>'white'];
+    $contactID = $this->individualCreate($contactParams);
+
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    // run importer
+    $importParams = [
+      'job_contract_id' => $contract->id,
+      'title' => 'test import role'
+    ];
+    $this->runImport($importParams);
+
+    $roleEntity = $this->findRoleByProperty($importParams['title'], 'title');
+    $this->assertEquals($importParams['title'], $roleEntity->title);
+  }
+
+  function testImportWithoutMandatoryFields() {
+    // run importer
+    $importParams = [
+      'title' => 'test import role'
+    ];
+    $this->runImport($importParams, 'assert_fail');
+  }
+
+  function testImportWithValidOptionValues() {
+    // create contact
+    $contactParams = ['first_name'=>'walter', 'last_name'=>'white'];
+    $contactID = $this->individualCreate($contactParams);
+
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    // run importer
+    $importParams = [
+      'job_contract_id' => $contract->id,
+      'title' => 'test import role',
+      'location' => 'amman',
+      'hrjc_region' => 'south amman',
+      'hrjc_role_department' => 'amman devs',
+      'hrjc_level_type' => 'guru'
+    ];
+    $this->runImport($importParams);
+
+    $roleEntity = $this->findRoleByProperty($importParams['title'], 'title');
+    $this->assertEquals($importParams['title'], $roleEntity->title);
+    $this->assertEquals($importParams['location'], $roleEntity->location);
+    $this->assertEquals($importParams['hrjc_region'], $roleEntity->region);
+    $this->assertEquals($importParams['hrjc_role_department'], $roleEntity->department);
+    $this->assertEquals($importParams['hrjc_level_type'], $roleEntity->level_type);
+  }
+
+  function testImportWithInvalidOptionValues() {
+    // create contact
+    $contactParams = ['first_name'=>'walter', 'last_name'=>'white'];
+    $contactID = $this->individualCreate($contactParams);
+
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    // run importer
+    $importParams = [
+      'job_contract_id' => $contract->id,
+      'title' => 'test import role2',
+      'location' => 'amman',
+      'hrjc_region' => 'southhggh ammandshhghg',
+      'hrjc_role_department' => 'amman devs',
+      'hrjc_level_type' => 'guru'
+    ];
+    $this->runImport($importParams, 'assert_fail');
```

Comment:
This test is a lot nicer than the old one, using the API. This way we know that the import feature should never allow invalid option values (even if they change this in the API). Thanks for this improvement!